tests/difference/core/quint_model/ccv.qnt

```diff
@@ -0,0 +1,828 @@
+            consumerStatus: Chain -> str,
+
+            // a monotonic strictly increasing and positive ID that is used
+            // to uniquely identify the Vscs sent to the consumer chains.
+            runningVscId: int,
+        }
+
+    // utility function: returns a provider state that is initialized minimally.
+    pure def GetEmptyProviderState(): ProviderState = 
+        {
+            chainState: GetEmptyChainState,
+            outstandingPacketsToConsumer: Map(),
+            receivedMaturations: Set(),
+            sentVscPacketsToConsumer: Map(),
+            providerValidatorSetChangedInThisBlock: false,
+            consumerStatus: Map(),
+            runningVscId: 0,
+        }
+        
+
+    // Defines the current state of a consumer chain. This information is accessible to that consumer on-chain.
+    type ConsumerState = {
+        // the state that each chain needs to store
+        chainState: ChainState,
+
+        // Stores the maturation times for VscPackets received by this consumer
+        maturationTimes: VscPacket -> Time,
+
+        // stores the received vscpackets in descending order of recency,
+        // i.e. newest first.
+        receivedVscPackets: List[VscPacket],
+
+        // Stores the list of packets that have been sent to the provider chain by this consumer
+        // and have not been received yet.
+        // ordered by recency, so the head is the oldest packet.
+        // In the implementation, essentially unacknowledged IBC packets.
+        outstandingPacketsToProvider: List[VscMaturedPacket],
+    }
+
+    // utility function: returns a consumer state that is initialized minimally.
+    pure def GetEmptyConsumerState(): ConsumerState = 
+        {
+            chainState: GetEmptyChainState,
+            maturationTimes: Map(),
+            outstandingPacketsToProvider: List(),
+            receivedVscPackets: List(),
+        }
+
+    // the state of the protocol consists of the composition of the state of one provider chain with potentially many consumer chains.
+    type ProtocolState = {
+        providerState: ProviderState,
+        // the state of each consumer chain.
+        // note that we assume that this contains all consumer chains that may ever exist,
+        // and consumer chains that are currently not running will have providerState.consumerStatus == NOT_CONSUMER or STOPPED.
+        consumerStates: Chain -> ConsumerState
+    }
+
+    // gets a protocol state that is initialized minimally.
+    pure def GetEmptyProtocolState(): ProtocolState = 
+        {
+            providerState: GetEmptyProviderState,
+            consumerStates: Map(),
+        }
+
+    type Error = {
+        message: str
+    }
+
+    // we return either a result or an error.
+    // if hasError is true, newState may be arbitrary, but the error will be meaningful.
+    // if hasError is false, error may be arbitrary, but newState will be meaningful.
+    type Result = {
+        hasError: bool,
```

Comment:
`hasError` is a bit redundant. I normally have just an `error` field of type `str`. If `error == ""`, then there is no error.

Reply:
Good point. I still like the explicitness of calling hasError, but I will put it just as a method of the result type and remove the field